heartbeat/monitors/active/http/http_test.go

```diff
@@ -808,6 +810,57 @@ func mustParseURL(t *testing.T, url string) *url.URL {
 	return parsed
 }
 
+// helper that compresses some content as gzip
+func gzipBuffer(t *testing.T, toZip string) *bytes.Buffer {
+	var gzipBuffer bytes.Buffer
+	gzipWriter := gzip.NewWriter(&gzipBuffer)
+	defer gzipWriter.Close()
+	_, err := gzipWriter.Write([]byte(toZip))
+	if err != nil {
+		t.Fatal(err)
+	}
+	return &gzipBuffer
+}
+
+func TestDecodesGzip(t *testing.T) {
+	gzip := gzipBuffer(t, "TestEncodingAccept")
+
+	server := httptest.NewServer(hbtest.CustomResponseHandler(gzip.Bytes(), 200, map[string]string{
+		"Content-Encoding": "gzip",
+	}))
+	defer server.Close()
+
+	evt := sendTLSRequest(t, server.URL, false, map[string]interface{}{
+		"response.include_body": "always",
+		"check.request.headers": map[string]interface{}{"Accept-Encoding": "gzip"},
+	})
+
+	content, err := evt.Fields.GetValue("http.response.body.content")
+
+	// decode gzip content if 'Content-Encoding' header is present
+	assert.NoError(t, err)
+	assert.Exactly(t, content, "TestEncodingAccept")
+}
+
+func TestNoGzipDecodeWithoutHeader(t *testing.T) {
```

Comment:
Can we clarify that this is the *response* header, not the request header

Reply:
https://github.com/elastic/beats/pull/28739/commits/af7937235ddb652ec3a2551b958a0aca1525cb2a